Determine callback URL instructions

WilliamLoy opened the pull request:
Using 3080 or 443 exclusively in instructions is confusing and can block progress if the user isn't using one of those ports. These instructions help determine the exact values that should be used.

WilliamLoy committed:
Determine callback URL instructions

Using 3080 or 443 exclusively in instructions is confusing and can block progress if the user isn't using one of those ports. These instructions help determine the exact values that should be used.

stevenGravy commented on docs/pages/access-controls/sso/google-workspace.mdx:
````diff
@@ -170,10 +170,25 @@ correct Google Workspace plan.
+You can lookup the ```public_addr``` of your Teleport proxy using the following command:
+
+```
+curl https://example.teleport.com/webapi/ping | jq '.proxy'
````
Bit of a chicken and egg thing.  This `curl` doesn't include a port like `3080` or `443`.  We should probably include the `3080` example.  I would also make this an include that can be reused.

WilliamLoy replied:
Will open a new PR for this

alexfornuto commented on docs/pages/access-controls/sso/google-workspace.mdx:
````diff
@@ -170,10 +170,25 @@ correct Google Workspace plan.
-In the [Create an OAuth client ID](https://console.cloud.google.com/apis/credentials/oauthclient) page of the GCP console, select "Web application" as the Application type, pick a name, then add the following as an authorized redirect URI.
+#### Determine your callback URL
+
+You'll need a callback URL in order to complete the creation of an OAuth client ID with Google. For most deployments, this URL is your ```public_addr``` value. This consists of a domain name and port number. 
````
```suggestion
You'll need a callback URL in order to complete the creation of an OAuth client ID with Google. For
most deployments, this URL is your `public_addr` value. This consists of a domain name and port number.
```

alexfornuto commented on docs/pages/access-controls/sso/google-workspace.mdx:
````diff
@@ -170,10 +170,25 @@ correct Google Workspace plan.
+
+You'll need a callback URL in order to complete the creation of an OAuth client ID with Google. For most deployments, this URL is your ```public_addr``` value. This consists of a domain name and port number. 
+
+You can lookup the ```public_addr``` of your Teleport proxy using the following command:
````
```suggestion
You can lookup the `public_addr` of your Teleport proxy using the following command:
```

alexfornuto commented on docs/pages/access-controls/sso/google-workspace.mdx:
````diff
@@ -170,10 +170,25 @@ correct Google Workspace plan.
+
+You can lookup the ```public_addr``` of your Teleport proxy using the following command:
+
+```
+curl https://example.teleport.com/webapi/ping | jq '.proxy'
````
````suggestion
```code
$ curl https://example.teleport.com/webapi/ping | jq '.proxy'
````

alexfornuto commented on docs/pages/access-controls/sso/google-workspace.mdx:
````diff
@@ -170,10 +170,25 @@ correct Google Workspace plan.
+curl https://example.teleport.com/webapi/ping | jq '.proxy'
+```
+
+Within the output of that command look for ```"public_addr": "example.teleport.com:443"```. Some cluster may specify a different port. Use the port you find in the output of the previous command for the next steps.
````
```suggestion
Within the output of that command look for `"public_addr": "example.teleport.com:443"`. Some
cluster may specify a different port. Use the port you find in the output of the previous command
for the next steps.
```

alexfornuto commented on docs/pages/access-controls/sso/google-workspace.mdx:
````diff
@@ -170,10 +170,25 @@ correct Google Workspace plan.
+
+Within the output of that command look for ```"public_addr": "example.teleport.com:443"```. Some cluster may specify a different port. Use the port you find in the output of the previous command for the next steps.
+
+#### Configure the OAuth Client ID
+In the [Create an OAuth client ID](https://console.cloud.google.com/apis/credentials/oauthclient) page of the GCP console, select "Web application" as the Application type, pick a name, then add the callback URL you gathered in the [Determine your callback URL](#determine-your-callback-url) section in the Callback URI field.
````
```suggestion
#### Configure the OAuth Client ID

In the [Create an OAuth client ID](https://console.cloud.google.com/apis/credentials/oauthclient)
page of the GCP console, select "Web application" as the Application type, pick a name, then add
the callback URL you gathered in the [Determine your callback URL](#determine-your-callback-url)
section in the Callback URI field.
```

WilliamLoy commented:
@ptgott my apologies, what is needed from me to move it forward?

WilliamLoy committed:
Update docs/pages/access-controls/sso/google-workspace.mdx

Co-authored-by: Alex Fornuto <alex.fornuto@goteleport.com>

WilliamLoy committed:
Update docs/pages/access-controls/sso/google-workspace.mdx

Co-authored-by: Alex Fornuto <alex.fornuto@goteleport.com>

WilliamLoy committed:
Update docs/pages/access-controls/sso/google-workspace.mdx

Co-authored-by: Alex Fornuto <alex.fornuto@goteleport.com>

WilliamLoy committed:
Update docs/pages/access-controls/sso/google-workspace.mdx

Co-authored-by: Alex Fornuto <alex.fornuto@goteleport.com>

WilliamLoy committed:
Update docs/pages/access-controls/sso/google-workspace.mdx

Co-authored-by: Alex Fornuto <alex.fornuto@goteleport.com>

WilliamLoy reviewed (commented):
The changes all look correct to me. I am electing to include the proxy port notes in a new PR since this iteration will solve what is currently a primarily incorrect instruction.

webvictim commented on docs/pages/access-controls/sso/google-workspace.mdx:
````diff
@@ -170,10 +170,32 @@ correct Google Workspace plan.
+You can lookup the `public_addr` of your Teleport proxy using the following command:
+
+```code
+$ curl https://example.teleport.com/webapi/ping | jq '.proxy'
````
You could actually just get the value here directly if you'd rather:
 
```suggestion
$ curl https://example.teleport.com/webapi/ping | jq -r '.proxy.public_addr'
```

webvictim commented on docs/pages/access-controls/sso/google-workspace.mdx:
````diff
@@ -170,10 +170,32 @@ correct Google Workspace plan.
+```
+
+Within the output of that command look for `"public_addr": "example.teleport.com:443"`. Some
+cluster may specify a different port. Use the port you find in the output of the previous command
````
```suggestion
clusters may specify a different port. Use the port you find in the output of the previous command
```

webvictim commented on docs/pages/access-controls/sso/google-workspace.mdx:
```diff
@@ -170,10 +170,32 @@ correct Google Workspace plan.
+You'll need a callback URL in order to complete the creation of an OAuth client ID with Google. For
+most deployments, this URL is your `public_addr` value. This consists of a domain name and port number.
+
+You can lookup the `public_addr` of your Teleport proxy using the following command:
```
```suggestion
You can lookup the `public_addr` of your Teleport proxy using the following command (replace `example.teleport.com` with the address of your cluster):
```

This is actually a bit of a chicken and egg problem because you need the public address of the cluster in order to get the public address of the cluster 😅

alexfornuto replied:
Agreed, unless there's a case where the public address of the proxy is not the same as the cluster name (I don't think there is).

Since this guide includes the tctl partial early, I think it's a safe assumption that at this point in the guide they know the proxy address.

webvictim replied:
There is, but it's far less common now that we're encouraging people to set their cluster name to an FQDN.

WilliamLoy replied:
I think the changes are meant to solve the lion's share of use cases, which I believe it does.

webvictim reviewed (approved):
Sorry for confusing things, I think these docs changes are useful to have!

WilliamLoy committed:
Switch from command example to static examples

webvictim commented on docs/pages/access-controls/sso/google-workspace.mdx:
```diff
@@ -170,10 +170,32 @@ correct Google Workspace plan.
+
+Most deployments use port 443 or port 3080. The callback URL you use might look like the following examples (replace example.teleport.com or example.teleport.sh with your cluster's value):
+
+Teleport Cloud: `example.teleport.sh:443/v1/webapi/oidc/callback`
```
```suggestion
Teleport Cloud: `https://example.teleport.sh:443/v1/webapi/oidc/callback`
```

WilliamLoy replied:
done

webvictim commented on docs/pages/access-controls/sso/google-workspace.mdx:
```diff
@@ -170,10 +170,32 @@ correct Google Workspace plan.
+
+Teleport Cloud: `example.teleport.sh:443/v1/webapi/oidc/callback`
+
+Teleport Enterprise: `example.teleport.com:443/v1/webapi/oidc/callback`
```
```suggestion
Teleport Enterprise: `https://example.teleport.com:443/v1/webapi/oidc/callback`
```

webvictim commented on docs/pages/access-controls/sso/google-workspace.mdx:
```diff
@@ -170,10 +170,32 @@ correct Google Workspace plan.
+
+Teleport Enterprise: `example.teleport.com:443/v1/webapi/oidc/callback`
+
+Teleport Enterprise: `example.teleport.com:3080/v1/webapi/oidc/callback`
```
```suggestion
Teleport Enterprise: `https://example.teleport.com:3080/v1/webapi/oidc/callback`
```

WilliamLoy replied:
done

alexfornuto reviewed (approved):
LGTM, pending @webvictim's outstanding suggestions.

WilliamLoy committed:
added suggestions

Commit:
Merge branch 'master' into google-idp-docs-port-fix

webvictim commented on docs/pages/access-controls/sso/google-workspace.mdx:
```diff
@@ -170,10 +170,32 @@ correct Google Workspace plan.
+
+Teleport Cloud: `https://example.teleport.sh:443/v1/webapi/oidc/callback`
+
+Teleport Enterprise: `https://example.teleport.com:443/v1/webapi/oidc/callback`
```
Yes, and I thought about making that edit too - I just thought it might be confusing for people to see no port vs 3080 on the next line. On the whole I would prefer to see no port as the browser is going to strip it from the URL the second it sees it anyway.

alexfornuto replied:
Well, if the context here is determining what URL to provide to Google to configure them as an IdP, then the determining factor should be if the field this URL is going into accepts https with or without the port specified (when it's default `443`).

WilliamLoy replied:
@zmb3 I am fine with no port, but the current example specifies 3080 without any additional context. I had a customer get stuck on this for a long time because of the 3080 example.

WilliamLoy replied:
@zmb3 I just removed the callback URL section and added a note to Configure the OAuth Client ID section on what the callback URL should be.

WilliamLoy committed:
collapsed determine callback URL instructions

I removed all port references and added a sentence to the end of the first paragraph of Configure the OAuth Client ID to determine the callback URL.

Commit:
Merge branch 'master' into google-idp-docs-port-fix

Commit:
Merge branch 'master' into google-idp-docs-port-fix

ptgott commented:
@WilliamLoy Just checking if you'd still like to merge this one. If so, please feel free to add it to the merge queue!

ptgott commented:
@alexfornuto @WilliamLoy Is this one good to go?

stevenGravy commented:
But we're still going to be showing `3080` then. We should update the example YAML to match


<img width="1064" alt="image" src="https://user-images.githubusercontent.com/60704961/233478041-a3431030-83ec-4017-b7e4-36b4fcfd76b3.png">

stevenGravy commented:
The Teleport Cloud tab should be updated to a teleport.sh example

WilliamLoy committed:
Merge branch 'gravitational:master' into google-idp-docs-port-fix

WilliamLoy committed:
Update gworkspace-connector-inline.yaml

WilliamLoy committed:
Update gworkspace-connector.yaml

WilliamLoy commented:
@stevenGravy do the example changes look good?

stevenGravy commented on examples/resources/gworkspace-connector-inline.yaml:
```diff
@@ -29,7 +29,7 @@ spec:
     }
   issuer_url: https://accounts.google.com
-  redirect_url: https://<cluster-url>:3080/v1/webapi/oidc/callback
+  redirect_url: https://<cluster-url>/v1/webapi/oidc/callback
```
Would make this `teleport.example.com` to match the earlier example

stevenGravy commented on examples/resources/gworkspace-connector.yaml:
```diff
@@ -17,7 +17,7 @@ spec:
   google_service_account_uri: file:///var/lib/teleport/gworkspace-creds.json
   issuer_url: https://accounts.google.com
-  redirect_url: https://<cluster-url>:3080/v1/webapi/oidc/callback
+  redirect_url: https://<cluster-url>/v1/webapi/oidc/callback
```
make this `mytenant.teleport.sh`

stevenGravy commented:
> @stevenGravy do the example changes look good?

PUt in some suggestions for the url names.  Apply if you can.

WilliamLoy commented:
Replaced by #25285

WilliamLoy commented:
@stevenGravy opened the new PR using a branch off master instead of a forked repo
